### DISCUSSION TITLE
Unable to login from Edge extension when FIFO 2FA enabled

### DISCUSSION THREAD

**Discussion by plinss** [dani-garcia/vaultwarden#2788] [Q&A]
### Subject of the issue
Not sure if this is an issue with Vaultwarden or the Bitwarden Edge extension. 

I just installed the extension in Edge and was unable to login to my vault.
In the extension, after entering the email address and master password, I get a page saying 'Refused to connect'. A bit later I get an email from Vaultwarden saying it saw a login but no 2FA. The Vaultwarden log shows: 'Error: 2FA token not provided'.

This account has both TOTP and FIDO 2FA enabled, and I'm able to login from the web client, desktop, and mobile apps.

I disabled FIDO 2FA for the account (leaving TOTP enabled) and was able to login (I got the TOTP prompt after entering the master password).
 

### Deployment environment

* vaultwarden version: 1.25.2 

* Install method: built from source

* Clients used: Edge extension 2022.9.1

* Reverse proxy and version: nginx 1.23.1

* MySQL/MariaDB or PostgreSQL version: PostgreSQL 13.8

* Other relevant details:

### Steps to reproduce

Enable FIDO and TOTP 2FA for account in web client. 
Install browser extension in Edge, try to login.

### Expected behaviour

Prompt for FIDO key or TOTP

### Actual behaviour

'Refused to connect' error message

**Comment by BlackDex** [comment]
Could you provide the support string which you can generate from the admin diagnostics page please.

Also, make sure you do not have added extra headers in nginx to block frames, or external content with custom CSP headers, those will break the functionality. Al the needed headers for security are configured in Vaultwarden already. Expect for HTST.

**Comment by plinss** [comment]
I was adding headers in nginx to prevent being framed, removed those and it works now. Sorry for the bother and thanks for the input.
